### DISCUSSION TITLE
Need (general?) help with module

### DISCUSSION THREAD

**Discussion by Kramsch** [johnnolan/encounter-stats#117] [Q&A]
Hello!

I wanted to use this module, but I'm running into some sort of problem. Maybe I just don't understand how to use the addon.

For Background Information:
DnD5e Version: 1.5.6
FVTT Encounter Stats Version: 0.21.1
Foundry Version: 9 build 238
Beyond20 companion module 1.4.0
Chrome Version: 97.0.4692.71
Beyond 20 Chrome Plugin Version: 2.6.0

If I start an encounter and roll some abilities from DnDBeyond (Beyond20),
after installing Beyond20 companion module attacks were shown. Without the Installation no attacks were shown in my fight entry. OK.
Kills are counted correctly, if its the players turn. Weapon/spell name with rolls are now shown correctly (not with this little bag and blue screen like in the readme, but normal as text format). Health is shown correctly.
Before installation of the companion module nothing was shown and only Health changes were written down.

**Actual Problem:**
But if I roll an attack and substract the total from my monster. The Min Damage, Max Damage and AVG Damage of this player will show NULL. Highest Average Damage will be set to NaN. Healing Actions, Highest Damage in 1 Hit, Most Damage overall and so on from the field above are not even counted.

**Question:**
Does anyone know if a How-To exists? I'm doing something wrong probably and want to see how I can do it better.

Best regards,
Kramsch

**Comment by johnnolan** [comment]
Hi, thanks for the question, really nicely detailed. I will try and get some time this week and have a play around with that setup locally. As soon as I figure it out I will tag the release here for you.

**Comment by Kramsch** [comment]
![image](https://user-images.githubusercontent.com/97339478/166438725-64c6f2f0-8f3e-4a57-861e-163e07e33644.png)

Hello There! John, could you find a way? I would be more than happy if we can solve this somehow.
I attached a screenshot of an test combat taken a few minutes ago after updating my modules (Beyond20 and FVTT Encounter Stats) and deactivating every other module I've installed.

I don't know if this would be relevant, but I use Linux as OS.

Here is my JSON Output, if it could help somehow. 
`{"encounterId":"Gq71uU9Zbqhw1Pio","round":2,"combatants":[{"name":"Stone Golem","id":"b2CwV8oroGkKi9Rl","tokenId":"a3iTlCuTxCRofTiR","img":"worlds/Esaroth/data/Bilder/Token-Automatic/stone_golem.Token.webp?1648376375452","type":"npc","hp":178,"max":178,"ac":17,"events":[{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"b2CwV8oroGkKi9Rl","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":0,"damageTotal":21,"item":{"name":"Slam","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"b2CwV8oroGkKi9Rl","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":27,"damageTotal":17,"item":{"name":"Slam","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"b2CwV8oroGkKi9Rl","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":26,"damageTotal":24,"item":{"name":"Slam","itemLink":null}}],"health":[{"id":null,"round":1,"tokenId":null,"actorId":"b2CwV8oroGkKi9Rl","max":178,"diff":9,"previous":178,"current":169,"isdamage":true,"isheal":false},{"id":null,"round":1,"tokenId":null,"actorId":"b2CwV8oroGkKi9Rl","max":178,"diff":12,"previous":169,"current":157,"isdamage":true,"isheal":false},{"id":null,"round":1,"tokenId":null,"actorId":"b2CwV8oroGkKi9Rl","max":178,"diff":8,"previous":157,"current":149,"isdamage":true,"isheal":false}],"kills":[],"summaryList":{"min":null,"max":null,"avg":null,"total":0},"roundSummary":{"individual":{},"totals":[]}},{"name":"Deekukra","id":"CT6VuDbNOGNTr5dS","tokenId":"6JHL6Nqe4TGq3hkA","img":"worlds/Esaroth/data/Bilder/Token-Automatic/kramsch_s_character.Token.webp?1641239872566","type":"character","hp":122,"max":122,"ac":20,"events":[{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":23,"damageTotal":9,"item":{"name":"Spear of Warning","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":20,"damageTotal":12,"item":{"name":"Spear of Warning","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":0,"damageTotal":0,"item":{"name":"Action Surge","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":19,"damageTotal":8,"item":{"name":"Spear of Warning","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":12,"damageTotal":6,"item":{"name":"Spear of Warning","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":0,"damageTotal":0,"item":{"name":"Second Wind","itemLink":null}},{"id":null,"actionType":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":0,"damageTotal":12,"item":{"name":"","itemLink":null}},{"id":null,"actionType":null,"round":2,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":0,"damageTotal":0,"item":{"name":"Quickened Healing","itemLink":null}},{"id":null,"actionType":null,"round":2,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","advantage":false,"isCritical":false,"isFumble":false,"disadvantage":false,"attackTotal":0,"damageTotal":7,"item":{"name":"","itemLink":null}}],"health":[{"id":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","max":122,"diff":17,"previous":122,"current":105,"isdamage":true,"isheal":false},{"id":null,"round":1,"tokenId":null,"actorId":"CT6VuDbNOGNTr5dS","max":122,"diff":24,"previous":105,"current":81,"isdamage":true,"isheal":false}],"kills":[],"summaryList":{"min":null,"max":null,"avg":null,"total":0},"roundSummary":{"individual":{},"totals":[]}}],"top":{"maxDamage":"Stone Golem<br />0","mostDamageInOneTurn":"None<br />undefined","highestAvgDamage":"Stone Golem<br />null","highestMaxDamage":"Stone Golem<br />null","mostKills":"Stone Golem<br />0","mostHealing":"Stone Golem<br />0","mostSupportActions":"Stone Golem<br />0","mostBattlefieldActions":"Stone Golem<br />0"},"templateHealthCheck":[]}`

EDIT: I replaced my screenshot with one after activating my modules again. Same look (except the date in title)

**Comment by Kramsch** [comment]
Okay. This can be closed. Have opened it as bug with "Report a bug" code
